tasks/install.js

```diff
@@ -0,0 +1,75 @@
+module.exports = function(grunt) {
+
+  /**
+   * Define tasks to quickly get Drupal running.
+   *
+   * grunt serve
+   *   Run Drupal via the PHP webserver and activate watch tasks.
+   * grunt serve:demo
+   *   Run Drupal via the PHP webserver without running watch tasks.
+   * grunt serve:test
+   *   Run Drupal via the PHP webserver without running watch tasks or opening
+   *   the browser.
+   */
+  grunt.loadNpmTasks('grunt-drush');
+
+  var Help = require('../lib/help')(grunt),
+    Drupal = require('../lib/drupal')(grunt);
+
+  var path = require('path'),
+    _ = require('lodash');
+
+  // If no path is configured for Drush, fallback to the system path.
+  var cmd = {cmd: Drupal.drushPath()};
+
+  var profile = grunt.config('config.serve.profile') || 'standard';
+
+  grunt.config(['drush', 'install'], {
+    args: ['site-install', '-yv', profile],
+    options: _.extend({
+      cwd: '<%= config.buildPaths.html %>'
+    }, cmd)
+  });
+  grunt.config(['drush', 'sql-connect'], {
+    args: ['sql-connect'],
+    options: _.extend({
+      cwd: '<%= config.buildPaths.html %>'
+    }, cmd)
+  });
+  grunt.config(['drush', 'sql-drop'], {
+    args:['sql-drop', '-y'],
+    options: _.extend({
+      cwd: '<%= config.buildPaths.html %>'
+    }, cmd)
+  });
+  grunt.config(['shell', 'loaddb'], {
+    command: 'gzip -dc <%= config.install.db %> | <%= config.install.dbConnection %>'
+  });
+
+  grunt.registerTask('install', 'Install Drupal with the configured profile, or from a database dump file if one exists.', function() {
+
+    // Check for a database file first.
+    var dbPath = grunt.config('config.install.db');
+    if (dbPath && grunt.file.exists(dbPath)) {
+      // @todo This doesn't work, the output of the sql-connect command is needed.
```

Comment:
I don't know how to dynamically capture the database credentials to use in the loaddb command below.